framework/meta/src/meta_cli.rs

```diff
@@ -0,0 +1,83 @@
+use super::{meta_config::MetaConfig, output_contract::OutputContractConfig};
+use crate::{
+    cli_args::{CliAction, CliArgs},
+    sc_upgrade::upgrade_sc,
+};
+use clap::Parser;
+use multiversx_sc::{
+    abi::ContractAbi, api::uncallable::UncallableApi, contract_base::ContractAbiProvider,
+};
+
+/// The ABI provider set when
+struct NoAbiProvider;
+
+impl NoAbiProvider {
+    pub const NAME: &str = "no-abi";
+}
+
+impl ContractAbiProvider for NoAbiProvider {
+    type Api = UncallableApi;
+
+    fn abi() -> ContractAbi {
+        ContractAbi {
+            name: Self::NAME,
+            ..Default::default()
+        }
+    }
+}
+
+fn process_abi<AbiObj: ContractAbiProvider>(cli_args: &CliArgs) -> Option<MetaConfig> {
+    let input_abi = <AbiObj as ContractAbiProvider>::abi();
+    if input_abi.name == NoAbiProvider::NAME {
+        return None;
+    }
+
+    super::meta_validate_abi::validate_abi(&input_abi).unwrap();
```

Comment:
Any particular reason why you're not importing the function instead of doing the fully qualified call?

Reply:
Also, is `unwrap` instead of `?` intentional?

Reply:
Changed to function import.

We are pretty top-level at this point, I felt thee was no need to pass errors even higher up.